backend/src/stwo/proof.rs

```diff
@@ -0,0 +1,22 @@
+use std::collections::BTreeMap;
+use stwo_prover::core::backend::BackendForChannel;
+use stwo_prover::core::channel::MerkleChannel;
+use stwo_prover::core::fields::m31::BaseField;
+use stwo_prover::core::poly::circle::CircleEvaluation;
+use stwo_prover::core::poly::BitReversedOrder;
+use stwo_prover::core::ColumnVec;
+
+/// For each possible size, the commitment and prover data
+pub type TableProvingKeyCollection<B, MC> = BTreeMap<usize, TableProvingKey<B, MC>>;
+
+pub struct TableProvingKey<B: BackendForChannel<MC>, MC: MerkleChannel> {
```

Comment:
One thing that is missing in this PR is serializing the proving key and setting it in the backend. Now `setup` is run when creating the backend. Is this something we'd do in a separate PR?

Reply:
Done